script/subtitles/imscsubtitles.js

```diff
@@ -4,28 +4,106 @@ define('bigscreenplayer/subtitles/imscsubtitles',
     'bigscreenplayer/domhelpers',
     'bigscreenplayer/debugger/debugtool',
     'bigscreenplayer/plugins',
-    'bigscreenplayer/utils/playbackutils'
+    'bigscreenplayer/utils/playbackutils',
+    'bigscreenplayer/utils/loadurl'
   ],
-  function (IMSC, DOMHelpers, DebugTool, Plugins, Utils) {
+  function (IMSC, DOMHelpers, DebugTool, Plugins, Utils, LoadURL) {
     'use strict';
-    return function (mediaPlayer, response, autoStart, parentElement, defaultStyleOpts) {
+    return function (mediaPlayer, captions, autoStart, parentElement, defaultStyleOpts, windowStartTime) {
```

Comment:
Suggestion: If you passed in mediasources you'd have access to `getWindowStartTime()` and also a reasonable place to put your segment calculation logic.

Reply:
Since currently subtitles is only interested in windowStartTime I feel it makes more sense to just pass it in directly.